grub-core/commands/blscfg.c

```diff
@@ -538,7 +538,7 @@ static int read_entry (
       char *separator;
 
       buf = grub_file_getline (f);
-      if (!buf)
+      if (!buf && grub_errno)
```

Comment:
Hi Matteo, 

help me to undestand: is the function `grub_file_getline` will set `grub_errno` in case of `EOF`?

Reply:
Not directly, but it calls `grub_file_read` which should.
I see the same pattern here:
https://github.com/rhboot/grub2/blob/fedora-41/grub-core/commands/legacycfg.c#L81-L84

Reply:
Can you try to reproduce the issue with the entry in the commit message?
Anyway, I don't know who designed `grub_file_getline` in a way which is impossible to distinguish between EOF and an empty line.
There are 14 usages for `grub_file_getline`, that could be changed.

Reply:
Seems a pretty old change though: 2cff3677bb8b21a7bb7ec97f1e142c42511e218a

Reply:
I don't know it if breaks other usages, what do you think about a change like 398f3466ce13cbc0bf4f2a132d6a119b55d5b1dc?

Reply:
@teknoraver thanks for the explanation. I have added other reviewers . Also, if possible try sending the same  patch but rebased for grub2-2.12 into the grub-devel mailing list.